uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/leave-request-popup-controller_test.js

```diff
@@ -0,0 +1,591 @@
+            $ctrl.onDateChange($ctrl.uiOptions.fromDate, true);
+            $ctrl.onDateChange($ctrl.uiOptions.toDate, false);
+            $scope.$digest();
+          });
+
+          it('balance is displayed', function () {
+            expect($ctrl.uiOptions.showBalance).toBeTruthy();
+          });
+        });
+      });
+
+      describe('when user cancels dialog (clicks X), or back button', function () {
+
+        beforeEach(function () {
+          $ctrl.cancel();
+        });
+
+        it('closes model', function () {
+          expect(modalInstanceSpy.dismiss).toHaveBeenCalled();
+        });
+      });
+
+      describe('leave absence types', function () {
+
+        describe('view all', function () {
+
+          beforeEach(function () {
+            $scope.$digest();
+          });
+
+          it('shows all items', function () {
+            expect($ctrl.absenceTypes.length).toBeGreaterThan(0);
+          });
+        });
+
+        describe('change selection', function () {
+          var beforeChangeAbsenceType, afterChangeAbsenceType;
+
+          beforeEach(function () {
+            beforeChangeAbsenceType = $ctrl.selectedAbsenceType;
+            $ctrl.selectedAbsenceType = $ctrl.absenceTypes[1];
+
+            $ctrl.onAbsenceTypeChange();
+            afterChangeAbsenceType = $ctrl.selectedAbsenceType;
+            $scope.$digest();
+          });
+
+          it('selects another absence type', function () {
+            expect(beforeChangeAbsenceType.id).not.toEqual(afterChangeAbsenceType.id);
+          });
+        });
+      });
+
+      describe('number of days selection', function () {
+
+        describe('multiple', function () {
+
+          it('selects by default', function () {
+            expect($ctrl.uiOptions.multipleDays).toBeTruthy();
+          });
+
+        });
+
+        describe('single', function () {
+
+          beforeEach(function () {
+            $ctrl.uiOptions.multipleDays = false;
+            $ctrl.changeInNoOfDays();
+            $scope.$digest();
+          });
+
+          it('selects single day', function () {
+            expect($ctrl.uiOptions.multipleDays).toBeFalsy();
+          });
+
+          it('hides to date and type', function () {
+            expect($ctrl.uiOptions.toDate).not.toBeDefined();
+            expect($ctrl.uiOptions.selectedToType).not.toBeDefined();
+          });
+
+          it('will reset balance and types', function () {
+            expect($ctrl.uiOptions.selectedFromType).not.toBeDefined();
+            expect($ctrl.uiOptions.selectedToType).not.toBeDefined();
+            expect($ctrl.balance.closing).toEqual(0);
+            expect($ctrl.balance.change.amount).toEqual(0);
+          });
+        });
+      });
+
+      describe('calendar', function () {
+
+        it('no dates are selected by default', function () {
+          expect($ctrl.uiOptions.fromDate).not.toBeDefined();
+          expect($ctrl.uiOptions.toDate).not.toBeDefined();
```

Comment:
Again, something that should be tested in the "initialization" block. Basically you should have one
```js
describe('when initialized', function () {

});
```